libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -393,6 +393,7 @@ void lua_scripts::run(void) {
     // panic should be hooked first
     if (setjmp(panic_jmp)) {
         if (!succeeded_initial_load) {
+            free(_heap);
```

Comment:
you would also need to set _heap to nullptr
what does this do on scripting restart?

Reply:
Scripting restart deletes the lua script object. https://github.com/ArduPilot/ardupilot/blob/b8690ba109531055fb235b0e7f20fefaabe61962/libraries/AP_Scripting/AP_Scripting.cpp#L237

But it was not first freeing the heap.

I was not setting nullptr because the object will get deleted after return.  

I think I will move to a destructor.